File: Java/Numbers/program54.java
//Accept number from user and count the number of factors 

import java.util.*;

class Number
{
    public int iNo;

    public Number(int x)
    {
        iNo = x;
    }

    public void Display()
    {
        System.out.println("The numbers is: "+this.iNo);
    }

    public int CountFactor()
    {
        int iCnt = 0;
        int iFactCnt = 0;

        for (iCnt = 2; iCnt <= (iNo/2); iCnt++)
        {
            if(iNo % iCnt == 0)
            {
                iFactCnt++;
            }
        }
        return iFactCnt;
    }
}

class program54
{
    public static void main(String arg[])
    {
        int iValue = 0;
        int iRet = 0;

        Scanner sobj = new Scanner(System.in);
        
        System.out.println("Enter the number");
        iValue = sobj.nextInt();

        Number obj = new Number(iValue);

        obj.Display();

        iRet = obj.CountFactor();
        System.out.println("Number of factors are: "+iRet);
    }
}

File: C++/Number/program54.cpp
// Accept number from user and display that number from 1 to that number in forward and reverse direction

#include <iostream>
using namespace std;

class Number
{
public:
    int iNo;

    Number(int x)
    {
        iNo = x;
    }

    void Display()
    {
        cout << "The number is: " << this->iNo << endl;
    }

    void DisplayF()
    {
        int iCnt = 0;

        cout<<"Forward Display"<<endl;

        for (iCnt = 1; iCnt <= iNo; iCnt++)
        {
            printf("%d\n", iCnt);
        }
    }

    void DisplayB()
    {
        int iCnt = 0;

        cout<<"Backard Display"<<endl;
        
        for (iCnt = iNo; iCnt >= 1; iCnt--)
        {
            printf("%d\n", iCnt);
        }
    }
};

int main()
{
    int iValue = 0;

    cout << "Enter the number" << endl;
    cin >> iValue;
        
    Number nobj(iValue);

    nobj.Display();

    nobj.DisplayF();
    nobj.DisplayB();

    return 0;
}
